src/main/java/org/candlepin/subscriptions/resource/api/v1/BillingAccountIdResource.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Red Hat, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * Red Hat trademarks are not licensed under GPLv3. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.subscriptions.resource.api.v1;
+
+import jakarta.transaction.Transactional;
+import jakarta.ws.rs.ForbiddenException;
+import java.util.ArrayList;
+import java.util.List;
+import lombok.extern.slf4j.Slf4j;
+import org.candlepin.subscriptions.db.HostTallyBucketRepository;
+import org.candlepin.subscriptions.db.model.BillingProvider;
+import org.candlepin.subscriptions.db.model.DbReportCriteria;
+import org.candlepin.subscriptions.resource.ResourceUtils;
+import org.candlepin.subscriptions.security.InsightsUserPrincipal;
+import org.candlepin.subscriptions.security.auth.ReportingAccessOrInternalRequired;
+import org.candlepin.subscriptions.tally.admin.api.SwatchTallyApi;
+import org.candlepin.subscriptions.tally.admin.api.model.BillingAccountId;
+import org.candlepin.subscriptions.tally.admin.api.model.BillingAccountIdResponse;
+import org.springframework.stereotype.Component;
+
+/** This resource is for exposing REST endpoints for Billing Account Ids. */
+@Component
+@Slf4j
+public class BillingAccountIdResource implements SwatchTallyApi {
+
+  private final HostTallyBucketRepository hostTallyBucketRepository;
+
+  @SuppressWarnings("java:S107")
+  public BillingAccountIdResource(HostTallyBucketRepository hostTallyBucketRepository) {
+    this.hostTallyBucketRepository = hostTallyBucketRepository;
+  }
+
+  @Transactional
+  @ReportingAccessOrInternalRequired
```

Comment:
The standard API implementation uses the "@ReportingAccessRequired" annotation, which allows access to the customer or RH associates with admin role as asked in the JIRA description.

With the new annotation, we're also allowing access to any RH associate regardless if it has admin role, try:

```
curl 'http://localhost:8000/api/rhsm-subscriptions/v1/instances/billing_account_ids?org_id=6261679' -H x-rh-identity:$(echo '{"identity":{"associate":{"Role":["any-role"],"email":"test@redhat.com","givenName":"Bill","surname":"Potato"},"auth_type":"saml-auth","type":"Associate"}}' | base64 -w0)
```
According to the JIRA description, this should return 403.

Reply:
This will be access controlled upstream (i.e. turnpike) such that an associate that reaches the endpoint will already be approved. 

See https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/134043

Reply:
For the monolith APIs we decided to just keep doing them the standard way without nginx-proxy. I'll update that card description. 

All routes are only accessible through turnpike however, even the ones using nginx-proxy will still go through turnpike first.